### DISCUSSION TITLE
Importing products from supplier-provided CSVs / how to implement new / used stock

### DISCUSSION THREAD

**Discussion by thejspro** [medusajs/medusa#5273] [Q&A]
Hi there, hope it's okay to post two questions in one thread.

I have two use-case scenarios, which I am not quite sure are supported by Medusa. Would really appreciate if someone with more knowledge could explain this to me please.

### Product import automation and poor quality stock lists

We receive stock lists from our suppliers that are all over the place... the only thing that's consistent are EAN codes, but the quality of other data is not great. So far we've been doing all data input manually, but it's a tedious task - we have to verify the EAN code, find the matching product, copy the necessary pieces of information and update if required.

I was hoping to utilise the product import feature or the Admin API to automate this process and make it a little bit better, but come to think about it, it's not as easy as it would seem.

Our suppliers typically only have some very basic information about the product, ie. EAN, colour, size and a simplified product name. Also, more often than not, even this information is formatted differently between our suppliers. So really, the only thing that's consistent, is the EAN code.

Can I lookup the product by the variant EAN code? If a match if found, I could simply fetch the existing data from our database.

But what about a scenario, in which we don't have the EAN code in the database yet? How would I go about adding such a product? I guess I would create a draft product with the basic information given by the supplier, but then I'd have to source the missing bits manually anyway, which makes the whole idea of automation kinda pointless.

Does anyone have a similar use case? Any ideas how to deal with it? Apologies, because I also realise it's not a strictly technical question.

### New / used product stock implementation

We also want to expand to selling second-hand products, but at the moment it seems that all variant EAN codes in Medusa must be unique and cannot have duplicates. This means that I am unable to add the same EAN code twice, because I get an error ("Error Product_variant with ean 1234567890 already exists.").

I was thinking about using a different field for this, for example the "Barcode" field, but that's less than ideal.

How can I add a product with the same EAN but a different attribute to Medusa? Here's an example of what I mean:

**Parent product**

```
T-shirt
```

**Product variants**

```
EAN: 1234567890
Name: Black T-shirt
Color: black
Condition: new
```

```
EAN: 1234567890
Name: Black T-shirt
Color: black
Condition: used
```

Apologies for the long post and will be keeping an eye out for updates. Thank you!

**Comment by srindom** [comment]
Regarding your supplier provided CSV files, you can customize the import strategy to match the format of the CSV file. See more here: https://docs.medusajs.com/development/batch-jobs/customize-import
